src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,156 @@
+
+            final Storage storage = storageDbRepository.getStorage(storageId)
+                    .orElseThrow(() -> new UnableProcessException("No storage with id: " + storageId));
+            final Timeline activeTimeline = getTimeline(eventType);
+            final TopicRepository currentTopicRepo =
+                    topicRepositoryHolder.getTopicRepository(activeTimeline.getStorage());
+            final TopicRepository nextTopicRepo = topicRepositoryHolder.getTopicRepository(storage);
+            final List<PartitionStatistics> partitionStatistics =
+                    currentTopicRepo.loadTopicStatistics(Collections.singleton(activeTimeline.getTopic()));
+
+            if (activeTimeline.isFake()) {
+                LOG.info("Switching from fake timeline to real one. Current topic {}", activeTimeline.getTopic());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, activeTimeline.getTopic(), new Date());
+                switchTimeline(eventType, nextTimeline, () -> nextTimeline.setSwitchedAt(new Date()));
+            } else {
+                final String newTopic = nextTopicRepo.createTopic(partitionStatistics.size(),
+                        eventType.getOptions().getRetentionTime());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, newTopic, new Date());
+                LOG.info("Switching timelines. Topics: {} -> {}", activeTimeline.getTopic(), nextTimeline.getTopic());
+                switchTimeline(eventType, nextTimeline, () -> {
+                    final Timeline.StoragePosition storagePosition =
+                            StoragePositionFactory.createStoragePosition(activeTimeline, currentTopicRepo);
+                    activeTimeline.setLatestPosition(storagePosition);
+                    nextTimeline.setSwitchedAt(new Date());
+                    timelineDbRepository.updateTimelime(activeTimeline);
+                });
+            }
+        } catch (final NakadiException ne) {
+            throw new TimelineException(ne.getMessage(), ne);
+        }
+    }
+
+    public Timeline getTimeline(final EventType eventType) throws TimelineException {
+        try {
+            final String eventTypeName = eventType.getName();
+            final Optional<Timeline> activeTimeline = eventTypeCache.getActiveTimeline(eventTypeName);
+            if (activeTimeline.isPresent()) {
+                return activeTimeline.get();
+            }
+
+            final Storage storage = storageDbRepository.getStorage(DEFAULT_STORAGE)
+                    .orElseThrow(() -> new UnableProcessException("Fake timeline creation failed for event type " +
+                            eventType.getName() + ".No default storage defined"));
```

Comment:
I don't see any place where DEFAULT_STORAGE is created in DB. We expect that someone will create it for us?